### ISSUE TITLE
lab 2

### ISSUE THREAD

**Issue by Varshith00** [Varshith00/aiml#2]
import pandas as pd
import numpy as np
import re
from sklearn.model_selection import train_test_split
from sklearn.feature_extraction.text import TfidfVectorizer
from sklearn.naive_bayes import MultinomialNB
from sklearn.metrics import accuracy_score, classification_report

data = {
    'Email_Content': [
        "Congratulations! You've won a $1000 Walmart gift card. Click to claim!",
        "Please confirm your account information to avoid suspension.",
        "Are you free for a meeting at 10 am tomorrow?",
        "Important update regarding your bank account.",
        "Exclusive offer just for you. Claim your prize now!",
        "Here is the report from last month's analysis.",
        "Friendly reminder: Meeting rescheduled to next Monday.",
        "Urgent: Your account has been compromised. Click here to secure it."
    ],
    'Label': [1, 1, 0, 1, 1, 0, 0, 1]  # 1 = Spam, 0 = Not Spam
}

df = pd.DataFrame(data)

def preprocess_text(text):
    text = re.sub(r'[^a-zA-Z\s]', '', text)
    return text.lower()

df['Email_Content'] = df['Email_Content'].apply(preprocess_text)

vectorizer = TfidfVectorizer(stop_words='english', max_features=50)
X = vectorizer.fit_transform(df['Email_Content'])
y = df['Label']

X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=0.3, random_state=42)

model = MultinomialNB()
model.fit(X_train, y_train)

y_pred = model.predict(X_test)
accuracy = accuracy_score(y_test, y_pred)
report = classification_report(y_test, y_pred, target_names=["Not Spam", "Spam"])

print("Model Accuracy:", accuracy)
print("\nClassification Report:\n", report)

feature_names = vectorizer.get_feature_names_out()
spam_influence = model.coef_[0]
spam_words = sorted(zip(spam_influence, feature_names), reverse=True)[:10]

print("\nTop words associated with spam:")
for influence, word in spam_words:
    print(f"{word}: {influence:.2f}")
